### PR TITLE
feat: add manager to department. 

### LINKED ISSUE #205: Add a manager by department.
- Add a manager by department, for this we can add an integer field in the department table. 
- [Admin-Panel] add a select field to assign a department to a user.

### PR DESCRIPTION
**Pull request by GeorginaFraire**
#### What does this PR do?
- add the field `manager_employee_id`
- add the relation between users and department
- Add the UI of the admin panel to show the departments 
- Add the UI to assign an admin to a department 
- Add the UI to add users to a department

#### How should this be manually tested?
1. Login as an admin.
2. Go to the panel admin
3. Select de department tab
4. Select the option to edit a department
5. Add an admin and save. 
6. Double check that the user selected appears as manager
7. Select the option to see the users.
8. Verify that you can see the users assigned to the department selected.
9. Remove one user and double check that the user disappears.
10. Add a user using the input search and add the user selected. 
11. Verify that the pagination and the search are working as expected. 

#### Any background context you want to provide?
As part of the new changes, we have to add the functionality to have a manager for each department.
For futures changes, we are going to use this manager to send an email by default when someone selects a department when creating a question.   

#### What are the relevant tickets?
Closes #205 

#### Screenshots
:camera:

[admin-panel]
Screen Shot 2023-04-21 at 17 48 40

[Department tab]
Screen Shot 2023-04-21 at 17 52 15

[Edit Department]
Screen Shot 2023-04-21 at 17 53 22

[User by department]

Screen Shot 2023-04-21 at 17 54 18

https://user-images.githubusercontent.com/97203617/233750790-e7dc791a-2b43-4ac2-9e39-7f498536d756.mp4

### CODE REVIEW

File: app/components/SearchDropdown/SearchDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/* eslint-disable dot-notation */
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as S from 'app/components/SearchDropdown/SearchDropdown.styled';
+
+// const SearchDropdown = React.forwardRef((ref, props) => {
```
Comment (Hoosep) [inline app/components/SearchDropdown/SearchDropdown.jsx:6]: Remove this code commented.

File: app/controllers/departments/list.js [hunk 2]
```diff
@@ -1,8 +1,55 @@
...
+
+const calOffset = (allPages, offset, limit, totalPages) => {
+  if (allPages) return 0;
+  if ((offset / limit) > totalPages) return 0;
+
+  return offset;
+};
+
+const listDepartments = async (params) => {
+  let page; let size; let search; let allPages;
```
Comment (Hoosep) [inline app/controllers/departments/list.js:26]: I think you can declare multiple variables like this:

```suggestion
  let page, size, search, allPages;
```
So you don't have to repeat multiple times the keyword _let_.

References:
https://linuxhint.com/javascript-declare-multiple-variables/#3

**Review by Hoosep — APPROVED** [review]
Hey Gina! I left you some comments. They are minor changes, so I'm gonna approve this PR now but please help me to address before merging.

Thanks! LGTM and thanks because of this effort

**Review by ManuelArechiga — APPROVED** [review]
LGTM :+1:.

**Review by Hoosep — APPROVED** [review]

**Pull request merged**